test/metric_value_benchmark/base_test.go

```diff
@@ -58,25 +60,32 @@ func (t *TestRunner) runAgent() (status.TestGroupResult, error) {
 		},
 	}
 
+	extraCommands := t.testRunner.getExtraCommands()
+	err := util.RunCommands(extraCommands)
```

Comment:
I think getExtraCommands() is too generic. It doesn't really specify what kind of commands you mean and when it will be used. How about replacing it with "SetupBeforeAgentRun", and instead of getting commands, just call util.RunCommands within Setup..() function?

Reply:
I love generic more than specific case. e.g `runExtraCommands` can run after or before CWA running and for after running CWA, it can stop `collectd`. I can create functions such as `SetupBeforeAgentRun` or `SetupAfterAgentRun` but its too hollow regarding the core functionality.

Reply:
But in the template though you only run this before CWA. Names should be easy to understand without new devs having to read where the functions will be used. It's unclear what these are 'extra' for. I think it's almost like naming things "doSomething()" Just a hidden closet to shove in everything that doesn't fit an existing stuff

Reply:
Hmm !!! Agree `Names should be easy to understand without new devs having to read where the functions will be used.` . Let's me think about how to refactor it a little bit more.